Newsletters: add 344 (2025-03-07) (#2211)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2025-03-07-newsletter.md

  project were linked in a [thread][poinsot pri] on Delving Bitcoin.  In
  the [first][poinsot pri1] blog post, Poinsot describes the benefits of
  long-term goal setting and the costs of ad hoc decision making.  In
  the [second][poinsot pri2] post, he argues that "Bitcoin Core should
  be a robust backbone for the Bitcoin network, balancing between
  securing the Bitcoin Core software and implementing new features to
  strengthen and improve the Bitcoin network."  In the [third][poinsot
  pri3] post, he recommends splitting the existing project into three
  projects: a node, a wallet, and a GUI.  This is within reach today
  thanks to the multiple-year effort of the multiprocess sub-project
  (see [Newsletter #39][news39 multiprocess] for our first mention of
  this sub-project in 2019).

  Anthony Towns [questions][towns pri] whether multiprocess would really
  allow an effective split, as the individual components would remain
  tightly coupled.  Many changes to one project would require also
  making changes to the other projects.  However, it would be a clear
  win to move features that don't currently require a node into a

>   library or tool that can be maintained independently.  He also

Author: Leaving as-is.  The motivation for this change is unclear to me.

  describes how some people are using nodes today with middleware that
  makes it easy for users to connect their wallets to their own personal
  nodes, although this is done using blockchain indexes (basically [a
  personal block explorer][topic block explorers])---something the
  Bitcoin Core project has previously rejected including in its node
  directly.  Finally, he [notes][towns pri2] that "to [him], providing
  wallet features (mostly) and a GUI (to a lesser extent[...]) is a way
  of keeping us honest to the principle of bitcoin being usable by a
  decentralized bunch of hackers, versus being something that you can
  only really use if you’re a whale or an established corporation
  willing to make a big investment."

  David Harding expresses [concern][harding pri] that refocusing the
  main project on just consensus code and P2P relay will make it harder
  for everyday users to use a full node to validate their own incoming
  wallet transactions.  He asks Poinsot and other contributors to
  consider focusing instead on making Bitcoin Core easier for everyday

>   users.  He describes the power of full node validation: those who

Reviewer: ```suggestion
  users.  He describes the power of full node validation: economically-relevant users who
```
Maybe this is pedantic, but operating a full node alone doesn't help to define the consensus rules. It does little good if I run a forgotten full node in my basement that I never interact with. (It does _some_ good because it filters invalid junk from its peers so they don't have to expend bandwidth and CPU to reject the junk themselves. But this is a small effect.)

The full nodes that _economic_ users run matter. They attach _value_ to coins of one chain (fork) over another, and their full nodes enable them tell the difference.

Author: I think the rest of the original sentence covered that point in a more general way: "...full nodes that validate the predominate amount of economic activity...".  The generality is required because _you_ don't need to be an economically relevant user if you operate the full node for people who are themselves economically relevant.  For example, if I operate the node that 10,000 highly active lightweight clients connect to, I don't need to perform any transactions myself in order to have significant economic relevance.

Reviewer: That's an excellent point; I hadn't considered the use of light clients.

  operate the full nodes that validate the predominate amount economic
  activity have the ability to define Bitcoin's consensus rules.  In an

>   example, he shows that even a 30 minute change in what rules are

Author: Good reference and another interesting way to think about it.  I hadn't seen that video before.

  process to activate the change in Bitcoin software.

  His post notes that "it’s only a guide to the cooperative path, where
  you make a change that makes everyone’s life better, and more or less
  everyone ends up agreeing that the change makes everyone’s lives
  better."  He also warns that "it's also only a fairly high-level
  guide."

- **Update on BIP360 pay-to-quantum-resistant-hash (P2QRH):** developer
  Hunter Beast [posted][beast p2qrh] an update on his research into
  [quantum resistance][topic quantum resistance] for [BIP360][] to the
  Bitcoin-Dev mailing list.  He's made changes to the list of
  quantum-secure algorithms he's proposing to use, is looking for
  someone to champion development of a pay-to-taproot-hash (P2TRH)
  scheme (see [Newsletter #141][news141 p2trh]), and is considering
  targeting the same security level as currently provided by Bitcoin
  (NIST II) rather than a higher level (NIST V) that requires more block
  space and CPU verification time.  His post received multiple replies.

- **Private block template marketplace to prevent centralizing MEV:** Matt
  Corallo and developer 7d5x9 [posted][c7 mev] to Delving Bitcoin about

>   allowing parties to bid in a public markets for selected space within

Reviewer: ```suggestion
  allowing parties to bid in public markets for selected space within
```
Or "a public market" if singular is better.

  miner block templates.  For example, "I’ll pay X [BTC] to include
  transaction Y as long as it comes before any other transactions which
  interact with the smart contract identified by Z".  This is something
  that transaction creators on Bitcoin already want for various
  protocols, such as certain [colored coin protocols][topic client-side
  validation], and it's likely to become even more desirable in the
  future as new protocols are developed (including proposals requiring
  consensus changes such as certain [covenants][topic covenants]).

  If the service of preferential transaction ordering within block
  templates is not provided by a trust-reduced public market, it's
  probable that it will instead be provided by large miners, who will
  compete with users of the various protocols.  This will require the
  miners obtain large amounts of capital and technical sophistication,
  likely leading to them earning significantly higher profits than smaller
  miners without those capabilities.  This will centralize mining and

>   allow the large miners to more easily censor Bitcoin transactions.

Reviewer: ```suggestion
  allow large miners to censor Bitcoin transactions more easily.
```

  The developers propose providing trust reduction by allowing miners to
  work on blinded block templates whose complete transactions aren't
  revealed to the miner until they've produced sufficient proof of work
  to publish the block.  The developers propose two mechanisms for
  achieving this without requiring any consensus changes:

  - **Trusted block templates:** a miner connects to a marketplace, selects
    the bids it wants to include in a block, and asks the marketplace to
    construct a block template.  The marketplace responds with a block

>     header, coinbase transaction, and partial merkle branch that allow

Reviewer: ```suggestion
    header, coinbase transaction, and partial merkle branch, allowing
```

Author: Keeping as-is.  I think both constructions are correct and the existing construction avoids the use of a comma shortly after a comma-delimited list, making the sentence easier to skim.

    the miner to generate proof of work for that template without
    learning its exact contents.  If the miner does produce the network
    _target_ amount of proof of work, it submits the header and coinbase
    transaction to the marketplace, which verifies the proof of work,
    adds it to the block template, and
    broadcasts the block.  The marketplace might include a transaction

>     paying the miner in the block template or it might pay the miner

Reviewer: ```suggestion
    paying the miner in the block template, or it might pay the miner
```

Author: Keeping as-is.  "It might pay" doesn't make sense here as an independent sentence, therefore it's not an independent clause, therefore this is a simple conjunction.

    separately at a later time.

  - **Trusted execution environments:** miners obtain a device with a
    [TEE][] secure enclave, connect to marketplaces and select bids they
    want to include in their blocks, and receive the transactions in
    those bids encrypted to the TEE's enclave key.  The block template

>     is constructed within the TEE and it provides the host operating

Reviewer: ```suggestion
    is constructed within the TEE, and provides the host operating
```

    system with the header, coinbase transaction, and partial merkle
    branch.  If the target proof of work is generated, the miner
    provides it to the TEE, which verifies it and returns the full
    decrypted block template for the miner to add to the header and broadcast.  Again, the block
    template might include a transaction paying the miner from a UTXO

>     belonging to the marketplace or the marketplace might pay the miner

Author: Same comment as above: https://github.com/bitcoinops/bitcoinops.github.io/pull/2211#discussion_r1984466124

  local commitment, potentially causing a race between the two transactions.
  Opting for the remote commitment transaction is beneficial to the node because
  it avoids `OP_CHECKSEQUENCEVERIFY` (CSV) [timelock][topic timelocks] delays
  and eliminates the need for additional transactions to resolve pending
  [HTLCs][topic htlc].

- [Eclair #3016][] introduces low-level methods for creating Lightning
  transactions in [simple taproot channels][topic simple taproot channels],
  without making any functional changes. These methods are generated with
  [miniscript][topic miniscript], and differ from those outlined in the [BOLTs
  #995][] specification.

- [LDK #3342][] adds a `RouteParametersConfig` struct that enables users to
  customize routing parameters for [BOLT12][topic offers] invoice payments.
  Previously limited to `max_total_routing_fee_msat`, the new struct now
  includes [`max_total_cltv_expiry_delta`][topic cltv expiry delta],
  `max_path_count`, and `max_channel_saturation_power_of_half`. This change
  brings the [BOLT12][] parameter setting in line with that of [BOLT11][].

- [Rust Bitcoin #4114][] lowers the minimum non-witness transaction size from 85

>   bytes to 62 bytes, aligning with Bitcoin Core’s policy (see Newsletter

Author: I think this "62" is a typo that should be "65".  Will make that change.
